include/wlr/types/wlr_output.h

```diff
@@ -303,26 +294,90 @@ void wlr_output_lock_attach_render(struct wlr_output *output, bool lock);
  * a lock.
  */
 void wlr_output_lock_software_cursors(struct wlr_output *output, bool lock);
+
 /**
- * Renders software cursors. This is a utility function that can be called when
- * compositors render.
+ * Hardware cursors:
+ *
+ * Hardware cursors are a feature of GPUs that allow a small framebuffer
+ * to be displayed over the main framebuffer. This is more efficient than
+ * drawing onto the main framebuffer using e.g. OpenGL.
+ *
+ * Not every backend/output will support hardware cursors, so you must always
+ * be prepared to fall back to rendering the cursor to the main framebuffer
+ * yourself.
+ *
+ * Hardware cursors are always in output buffer coordinates, and does not take
+ * scale nor rotation into account. You are responsible for doing this
+ * yourself.
+ *
+ * Hardware cursors are always in the ARGB8888 format.
  */
-void wlr_output_render_software_cursors(struct wlr_output *output,
-	pixman_region32_t *damage);
 
+/**
+ * Sets the size of the hardware cursor buffer to the values pointed to by
+ * x and y.
+ *
+ * Some backends have strict requirements for the cursor size and may not be
+ * able to use the size you requested exactly. The actual value used will be
```

Comment:
Are these strict requirements a min/max value, or do they get more complex (e.g. must be even)? If the former, I'd rather expose the min/max and let the compositor munge the numbers themselves, then fail opaquely if they don't meet the constraints.

Reply:
On DRM, this indicates a width/height that works. This doesn't mean that it's the min/max: for instance, sizes too small might fail. There's no way to get the min/max that i know of.

Compositors should just use this size if the backend has such limitation.